subworkflows/local/profiling.nf

```diff
@@ -364,24 +365,71 @@ workflow PROFILING {
                                 .filter{
                                     if (it[0].is_fasta) log.warn "[nf-core/taxprofiler] mOTUs currently does not accept FASTA files as input. Skipping mOTUs for sample ${it[0].id}."
                                     !it[0].is_fasta
                                 }
-                                .multiMap {
-                                    it ->
-                                        reads: [it[0] + it[2], it[1]]
-                                        db: it[3]
+                                .branch {
+                                    longread: it[0].instrument_platform == 'OXFORD_NANOPORE'
+                                    shortread: it[0].instrument_platform != 'OXFORD_NANOPORE'
                                 }
+        ch_input_for_motus_longread = ch_input_for_motus.longread
+                                        .multiMap {
+                                            it ->
+                                                reads:[it[0] + it[2], it[1]]
+                                                db: it[3]
+                                        }
+        ch_input_for_motus_shortread = ch_input_for_motus.shortread.view()
+                                        .multiMap {
+                                            it ->
+                                                reads:[it[0] + it[2], it[1]]
+                                                db: it[3]
+                                        }
+
+
+        MOTUS_PREPLONG ( ch_input_for_motus_longread.reads, ch_input_for_motus_longread.db )
+
+        ch_database_for_motus = databases
+                                    .filter { meta, db -> meta.tool == 'motus' }
+                                    .map { meta, db -> [meta.db_name, meta, db] }
+
 
-        MOTUS_PROFILE ( ch_input_for_motus.reads, ch_input_for_motus.db )
+        ch_prepped_to_motus = MOTUS_PREPLONG.out.out
+                                    .map { meta, reads -> [meta.db_name, [meta, reads]] }
+                                    .combine(ch_database_for_motus, by:0)
+                                    .map { item ->
+                                        def db_name = item[0]
+                                        def meta = item[1][0]
+                                        def reads = item[1][1]
+                                        def db_meta = item[2]
+                                        def db = item[3]
+                                        def meta_keys = ['id', 'run_accession', 'instrument_platform', 'single_end', 'is_fasta', 'type']
+                                        def new_meta = meta.subMap(meta_keys)
+
+                                        def db_meta_keys = db_meta.keySet()
+                                        def new_db_meta = db_meta.subMap(db_meta_keys) + [type: meta.type]
+
+                                        [new_meta, [reads], new_db_meta, db]
+                            }
```

Comment:
Has the langauge sderver autoformatting been run on this file? The indentation looks all weird